client/user/delete_test.go

```diff
@@ -0,0 +1,48 @@
+package user_test
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/CiscoDevnet/terraform-provider-cdo/go-client/internal/http"
+	"github.com/CiscoDevnet/terraform-provider-cdo/go-client/user"
+	"github.com/jarcoal/httpmock"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestDelete(t *testing.T) {
```

Comment:
nit: maybe use the same structure as the other tests?

Reply:
I found it very hard to read the other tests, and in the case of the create and update tests, I found that the whole assertFunc approach made it impossible for me to do any tests other than nil check tests (rather than testing that the value returned was expected), which is why I did it this way.